clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -451,7 +453,7 @@ boolean hasProducerIdAndEpoch(long producerId, short producerEpoch) {
     /**
      * Set the producer id and epoch atomically.
      */
-    void setProducerIdAndEpoch(ProducerIdAndEpoch producerIdAndEpoch) {
+    private void setProducerIdAndEpoch(ProducerIdAndEpoch producerIdAndEpoch) {
```

Comment:
Reducing visibility here to make sure that test cases are forced through the proper state transitions.